.travis.yml

```diff
@@ -0,0 +1,46 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+language: node_js
+node_js:
+  - "12"
+
+git:
+  depth: 1
+
+services:
+  - 'docker'
+
+before_script:
+  - npm install -g @angular/cli
+  - npm install karma
+  - npm install -g protractor
+  - npm install yarn
+
+jobs:
+  include:
+    - stage: pre-commit checks
+      script:
+        - make check-license
+        - make build
```

Comment:
@kingamarton  @yangwwei  @wilfred-s 
We could add a make command which will do linting, run unit tests, prod build and prepare final image using docker. This will avoid the need of before_script in the travis file.